runtime/lua/vim/lsp.lua

```diff
@@ -307,6 +307,241 @@ local function create_and_initialize_client(config)
   return client.id, nil
 end
 
+local client_config_list_keys = {
+  filetypes = true,
+  on_attach = true,
+  on_exit = true,
+  on_init = true,
+  root_markers = true,
+}
+
+--- Similar to `vim.tbl_deep_extend('force', ...)`, except that lists specified in
+--- `client_config_list_keys` are extended instead of replaced.
+--- @param ... vim.lsp.Config
+--- @return vim.lsp.Config
+local function merge_configs(...)
```

Comment:
Just for reference: is this similar to the `vim.merge` behavior? https://github.com/neovim/neovim/pull/15124

Reply:
Not really, and it won't be possible to generalize without some kind of schema since certain keys have specific merge semantics. E.g. `on_attach` is coerced into a list, whereas `cmd` is always replaced.

Reply:
(We discussed adding some schema to `tbl_deepextend` but quickly dismissed it as infeasible in general, especially given the rather academic -- at that time -- use cases.)